com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/util/NameUtil.java

```diff
@@ -122,8 +122,11 @@ public static String getApplicationName(Module module) {
     }
 
     public static String getServiceName(Resource resource) {
+        return getServiceNameOrDefault(resource, null);
+    }
+    public static String getServiceNameOrDefault(Resource resource, String defaultServiceName) {
         return (String) resource.getParameters()
-                                .get(SupportedParameters.SERVICE_NAME);
+            .getOrDefault(SupportedParameters.SERVICE_NAME, defaultServiceName);
```

Comment:
There should be no need for this default. It should be set every time in `CollectSystemParametersStep`.